### DISCUSSION TITLE
Can pyperplan plan with domains that use metrics ?

### DISCUSSION THREAD

**Discussion by jorgerando** [aibasel/pyperplan#20] [Q&A]
Hello, I'm trying to make plans for domains that use PDDL fluents like this :

(define (domain tequila)

  (:requirements :typing :fluents)

  (:types bottle )

  (:functions
    (amount ?b - bottle)
    )
  (:predicates
    (bottle-finished ?b - bottle)
    )

  (:action drink
  :parameters (?b - bottle)
  :precondition (>= (amount ?b) 1)
  :effect (decrease (amount ?b) 1)
  )
 (:action done-drinking
 :parameters (?b - bottle)
 :precondition (= (amount ?b) 0)
 :effect (bottle-finished ?b)
 )
)

Can pyperplan plan on these types of domains ?

**Comment by maltehelmert** [comment]
No, this is "numerical planning" and goes beyond the "classical planning" fragment that Pyperplan supports.

**Reply by jorgerando** [reply]
Hello, thank you very much for the information

I don't know if the question is a bit stupid, excuse my ignorance

Do you know of any numerical planner that allows you to implement custom heuristics like in Pyperplan ?

**Reply by maltehelmert** [reply]
I don't, but perhaps others that are following this do? If not, a good place to ask this question is the "planning community" slack server. It is linked from here: http://planning.domains/about

**Comment by jendrikseipp** [comment]
I know that https://sites.google.com/view/enhsp/ supports heuristics, but it's not implemented in Python. I'd also recommend the slack server.

**Comment by meneguzzi** [comment]
Hi,

	I don’t know if absolutely need Python, but check the ENHSP planner: https://gitlab.com/enricos83/ENHSP-Public <https://gitlab.com/enricos83/ENHSP-Public>

Best wishes,
--
Felipe Meneguzzi
***@***.***

**Comment by meneguzzi** [comment]
Jendrik beat me to it.
--
Felipe Meneguzzi
***@***.***
